[6.3 Feature] Canned Role Org Admin feature UI Stubs

jyejare opened the pull request:
$subject

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -567,30 +567,42 @@ def test_positive_create_overridable_filter(self):
-        in role
+    @tier1
+    def test_negative_update_role_taxonomies_overridden_filters(self):
+        """Overridden role filters taxonomies doesnt updates on update of role
```
`Overridden role filters taxonomies doesnt updates on update of role taxonomies` is not readable enough from language perspective

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -567,30 +567,42 @@ def test_positive_create_overridable_filter(self):
+    @stubbed
+    @tier2
+    def test_positive_override_checkmark(self):
+        """Filter table displays checkmark image in override column for
```
Same here. I can understand steps and expected results, but not description

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
-
-        :expectedresults:
+            1. create Org Admin by cloning 'Organization admin' role which has
+                most resources permissions
```
four spaces are needed

jyejare replied:
Didnt get that... Where do we need 4 spaces ?

oshtaier replied:
sorry for that

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
+    @tier1
+    def test_positive_create_org_admin_from_clone(self):
+        """Create Org admin role which has access to most of the resources
+            within organization
```
don't think that spaces are needed here

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
         :steps:
 
-            1. Attempt to clone any existing role(e.g Manager Role)
+            1. Create Org Admin by cloning 'Organization admin' role
```
`Org Admin` is a name or what?

jyejare replied:
The org admin isnt a role, but we will create a role that will have all the permissions for perticular org(s) and location(s).
In short, Org Admin is shortname for Admin that controls the Org.

jyejare replied:
You can create org admin by cloning 'Organization admin' role, and assign org and location to that cloned role on which the user will have control.

jyejare replied:
To create an Org Admin both the steps in this test is required.

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
+    @tier3
+    def test_positive_access_entities_from_org_admin(self):
+        """User can access resources within its taxonomies if assigned role
+            has permission for same taxonomies
```
no spaces are needed here

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
+
+        :steps:
+
+            1. Create org_admin and assign taxonomies to it
```
What is `org_admin` here?

jyejare replied:
Same as above explained org admin, I am renaming it to have the same nomenclature.

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
+        :steps:
+
+            1. Create org_admin and assign taxonomies to it
+            2. Create user with same taxonomies as above org admin taxonomies
```
`Create user with same taxonomies as role above` sounds better.

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
+
+            1. Create org_admin and assign taxonomies to it
+            2. Create user with same taxonomies as above org admin taxonomies
+            3. Assign above cloned role to user.
```
`Assign cloned role to user above` sounds better.

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
+    @tier3
+    def test_negative_access_entities_from_org_admin(self):
+        """User can not access resources in taxonomies assigned to role if
+            its own taxonomies are not same as its role
```
no spaces needed

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
-            1. On cloning, taxonomies of the overridden filters in cloned role
-                are set to None
+            1. Taxonomies of the 'parent role overridden filters' are set to
+                None in cloned role
```
probably, 4 spaces neeeded

jyejare replied:
I didnt understand this but make test-docstrings doesnt throw any exception for this

oshtaier replied:
nevermind )

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -775,23 +812,493 @@ def test_positive_empty_filter_taxonomies_in_cloned_role(self):
+    @stubbed
+    @tier2
+    def test_negative_assign_taxonomies_by_org_admin(self):
+        """Org Admin doesnt have permissions to assign org/loc to any of
```
`Org Admin doesn't have permissions to assign org/loc to any of its entities`

oshtaier commented on tests/foreman/ui/test_role.py:
```diff
@@ -775,23 +812,493 @@ def test_positive_empty_filter_taxonomies_in_cloned_role(self):
+    @tier2
+    def test_positive_taxonomies_control_to_superAdmin_with_org_admin(self):
+        """Super Admin can access entities in taxonomies assigned to Org Admin
+        in org admin's presence
```
can you please clarify word `presence` here?

jyejare replied:
Presence in the sence, Super Admin can access the entities in Org Admin taxonomies even though the Org Admin present(exists/not_deleted). 
In short what I want to test is - > The control of taxonomies are not completely taken by Org Admin from Super Admin, Super Admin can still access what Org Admin can access
Simply removing the presence

jyejare commented:
Hey @oshtaier I fixed all the necessary comments and replied on your questions.
Also I have added 5 new testcases for LDAP authantication in bottom of the file.

Please revisit again, Thanks.

oshtaier commented:
Sorry man for delays from my side. It is just too many of PRs at that moment

jyejare commented:
@oshtaier , I updated all the tier2 and tier3 tests with case levels.

sghai commented on tests/foreman/ui/test_role.py:
```diff
@@ -660,138 +658,827 @@ def test_positive_disable_all_filters_overriding_option(self):
+
+    @stubbed
+    @tier2
+    def test_positice_access_users_inside_org_admin_taxonomies(self):
```
positive ?

sghai commented:
Looks great w/ so many test scenarios. Please fix the typo and PR is good.

sghai commented:
ACK

oshtaier commented:
Good, if that PR accumulated all comments from other Canned Role PRs then good to go
ACK pending @sghai comment

jyejare commented:
@sghai I fixed your comment, please merge.
